modules/foreman_compute_profile.py

```diff
@@ -82,38 +165,76 @@
     'name': 'name',
 }
 
+compute_attribute_name_map = {
+    'compute_resource': 'compute_resource',
+    'vm_attrs': 'vm_attrs',
+}
 
-def main(module):
-
-    (module_params, state) = module.parse_params()
-    name = module_params.get('name')
-    updated_name = module_params.get('updated_name')
-
-    module.connect()
-
-    data = {
-        'name': name
-    }
 
-    compute_profile = cement.find_compute_profile(module, name=name, failsafe=True)
+def compute_attribute(module, compute_profile, attributes):
+    # If we don't work on a copy, we hit a TypeError: Value of unknown type: <class 'nailgun.entities.ComputeProfile'> exception
+    # from module.exit_json(changed=changed) method
+    compute_attribute_dict = attributes.copy()
+    for key in ['compute_resource', 'vm_attrs']:
+        if key not in compute_attribute_dict:
+            module.fail_json(msg='compute_attribute must have %s.' % key)
 
-    if state == 'present' and updated_name:
-        data['name'] = updated_name
+    try:
+        compute_attribute = find_compute_attribute(module, compute_profile_name=compute_profile.name,
+                                                   compute_resource_name=compute_attribute_dict['compute_resource'], failsafe=True)
+        compute_attribute_dict['compute_resource'] = find_compute_resource(module, name=compute_attribute_dict['compute_resource'], failsafe=False)
+    except Exception as e:
```

Comment:
Why do `try` - `except` here, when you could just do `failsafe=False` (which is the default iirc) ?

Reply:
It's only to catch any other exceptions  than "find failed for entity"  that can happen when we search an entity. failsafe only take care of returning `None` if we don't find any entity instead of failing.

In this case, if we cannot find compute attribute, we will create it, but if we don't find compute resource, then it's an error and we want to stop execution now (failsafe=false). I think the `try - except` is needed here, but yes as `failsafe=False` by default, I can remove it if you want.

Reply:
OK, i do not know what 'every other exception' we might expect here, but i see that pattern all over the place in the other modules. So i am fine with that here.
We should however think about whether we consider those exceptions worth handling at all, or DRYing them into the find_* methods. But that should not block this PR.
@evgeni , @sean797 , @manuelbonk?

Reply:
Yeah, let's not block here. It's a common pattern at the moment and we might DRY it up later.